airgun/widgets.py

```diff
@@ -2124,44 +2124,43 @@ def fill(self, value):
 
 
 class PopOverWidget(Widget):
-    """Popover-content UI widget which contains header, drop_down, input_box, or textarea and
+    """Popover-content UI widget which contains header, drop_down, input_box/textarea and
     submit button. This is associated within a table.
 
     Example html representation::
 
-        <h3 class="popover-title">Authorize login delegation (Default: false)</h3>
-            <form class="form-inline editableform" style="">
-                <select class="form-control input-sm">
-                    <option value="0">No</option>
-                    <option value="1">Yes</option></select>
-                <div class="editable-buttons">
-                    <button type="submit" class="btn btn-primary btn-sm editable-submit">
-
-        <h3 class="popover-title">Authorize login delegation auth ...(Default: Not set)</h3>
-            <form class="form-inline editableform" style="">
-                <input type="text" ...>
-                <div class="editable-buttons">
-                    <button type="submit" class="btn btn-primary btn-sm editable-submit">
+        <div class="modal-content" role="document">
+            <h4 class="modal-title">Update value for Connect by IP setting</h4>
+            <form class="form-horizontal well">
+                <select name="value" class="form-control">
+                    <option value="true">Yes</option>
+                    <option value="false">No</option>
+                <div class="form-actions">
+                    <button type="submit" class="btn btn-primary">
+
+        <div class="modal-content" role="document">
+            <h4 class="modal-title">Update value for Administrator email address setting</h4>
+            <form class="form-horizontal well">
+                <input name="value" class="form-control" value="root@rhts.example.com">
+                <div class="form-actions">
+                    <button type="submit" class="btn btn-primary">
 
     Locator example::
 
-        //div[contains(@class, 'editable-open')]
+        //div[contains(@class,'modal-content')]
     """
 
-    ROOT = '.'
-    column_value = Text(".//span[contains(@class, 'editable-click')]")
```

Comment:
What is the reason for removing `column_value` from `PopOverWidget` and adding it to `SettingsView`? Note that this breaks `PopOverWidget.read()`:

```
(Pdb) view.table.row()['Value'].widget.read()
*** AttributeError: 'PopOverWidget' object has no attribute 'column_value'
```
Can you instead keep `column_value` in `PopOverWidget` and implement a `click()` method in it if necessary?

Reply:
As a followup, I think I see the reason for moving `column_value`, as part of changing `PopOverWidget.ROOT` from `'.'` to `"//div[contains(@class,'modal-content')]"`. I think a better solution would be to keep the original `ROOT` and `column_value` in `PopOverWidget`, and move the widgets associated with the modal window that pops up to a separate view, e.g.,

```
class PopOverModalView(View):
    ROOT = "//div[contains(@class,'modal-content')]"
    header = Text(".//h4[contains(@class, 'modal-title')]")
    input_box = TextInput(locator=".//input[contains(@class, 'form-control')]")
    drop_down = Select(".//select[contains(@class,'form-control')]")
    submit = Text(".//button[@type='submit']")
```

`PopOverWidget.fill()` would then click on `column_value`, then instantiate `PopOverModalView` and access the modal's widgets through the view.

Reply:
@mshriver @omkarkhatavkar @tpapaioa @latran I have made changes as per this comment and it seems to be working. But I am facing new issue, when I try to update a setting new value is being appended to old one. 
Can someone take this up?